### ISSUE TITLE
[bug] dotdrop v0.26.1 deleted my ~/.config/

### ISSUE THREAD

**Issue by TheNetAdmin** [deadc0de6/dotdrop#174]
Dotdrop version (and git commit if run from source): v0.26.1
Using dotdrop: from pip

**Describe the bug**

In config file, I set `my/config` as `link_children` to `~/.config`

It worked fine before, but today it requested to delete `~/.config` to create link, and I didn't notice it until I already said yes.

So my `~/.config` is gone, it's in root partition and I cannot get physical touch to the machine in the next few months. So I cannot find them back. Sad story.

After that, it even asked me to delete my `~` to create link, and fortunately, I didn't say 'yes'.

**My thoughts**

~~I don't know which part is wrong and I don't think I can reproduce it on other machines.~~ **I found out the reason, please see the following comments.**

But what matters is to prevent these bugs **as a user**, not a developer:

**Please disable the permission to delete your important directories, especially those specified in dotdrop config file!** I checked `sudo chattr +a .config` and it seems to work, I cannot delete `~/.config/` but I can append contents to the files under `~/.config/`. It's not perfect and I'm still trying to find out better ways to protect a folder but not its content. `chattr +u .config` may work on xfs, but not ext4, according to `chattr`'s manpage.

Or other ways maybe sync the `.config` periodically, or make `.config` a link instead of a normal dir.

Fortunately, I'm not using that machine's desktop, and I planned to install a new Linux on a new SSD drive when I go back, so eventually, I'll get rid of those configs. And all my cross-machine configs are properly stored at GitHub, so I didn't lose much.

But just in case, **do protect your important directories from any kind of accidental deletions!**

**Comment by TheNetAdmin** [comment]
Anyone may close the issue, as I didn't find the bug. This issue is just a tip for the users.

If anyone can put this tip into README or wiki, I'd be really appreciated.

**Comment by davla** [comment]
I honestly understand the magnitude of this problem. Errors in `config.yaml` files can happen, especially when learning to use dotdrop. And it can also be that someone is just getting started with dotfiles management and doesn't have a git repo setup yet, which means loss of data.

Thus, I feel like `$HOME` and `$HOME/.config` deserve a special treatment. They are very likely to be involved in dotfiles management, very unlikely to be managed as a whole, and deleting them has catastrophic consequences.

I suggest exiting with an error when dotdrop would destroy their content, for example when they are about to be replaced by a symbolic link. And a flag to disable this behavior *if it's really what you want*, something like `--unsafe`.

**Comment by TheNetAdmin** [comment]
I think I found out why:

I have 2 machines to sync, the new one with v0.29.3, the old one with v0.26.1. Originally the new machine used v0.26.1 too, but I just upgraded it to v0.29.3.

After the upgrade, it silently changed my config file to use new option: from `link: link_children` to `link_children: true`. It didn't notice me, and I pushed it to git, pulled it on my old machine, and try to use v0.26.1 to execute this new config file.

**But v0.26.1 doesn't recognize the new option, so it misunderstood and offered to delete my dirs!**

So please, next don't upgrade the config file without a serious warnning, we may use different versions of dotdrop on different machines.

**Comment by deadc0de6** [comment]
Thanks for reporting this. I feel sorry for you, hope you are able to recover your file :-s
I'll implement both recommended changes:
* request for a specific flag or simply refuse to remove `$HOME` or `$HOME/.config` and maybe some other very important files
* add a minimum supported version field in the config file

I'll keep you updated!

**Comment by deadc0de6** [comment]
I have added more checks to avoid this in the future:

* (1) dotdrop now refuses to remove `$HOME` or `$HOME/.config`
* (2) dotdrop will print a warning at the end of any operation if the config file was updated (due for example to deprecated entries automatically updated)
* (3) from now on a `minversion` entry will be present in the config file

(3) unfortunately won't completely solved issues of people using versions of dotdrop where the `minversion` wasn't implemented but at least (1) and (3) will warn the user of a change and avoid erasing those two critical directories.

Note that the above won't happen if dotdrop is used as a submodule (which is the recommended way) as dotdrop will automatically be updated at each run.

I'm again sorry for what happened to you and hope you were able to recover your files. Hopefully at least above fixes will avoid this in the future.
